### ISSUE TITLE
“Illegal reflective access by com.jogamp.common.os.NativeLibrary” when using P3D (caused by JOGL)

### ISSUE THREAD

**Issue by sampottinger** [benfry/processing4#22]
Filing for [kll](https://discourse.processing.org/u/kll) from a [processing forums discussion](https://discourse.processing.org/t/processing-in-style-with-java-11/13776/24). The following happens when using P3D:

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.jogamp.common.os.NativeLibrary$3 (file:/.../core/library/gluegen-rt.jar) to method java.lang.ClassLoader.findLibrary(java.lang.String)
WARNING: Please consider reporting this to the maintainers of com.jogamp.common.os.NativeLibrary$3
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

**Comment by sampottinger** [comment]
This comes from JOGL itself and is related to [JEP 261](https://openjdk.java.net/jeps/261) as tracked at https://jogamp.org/bugzilla//show_bug.cgi?id=1363. There is no resolution on the Processing side except for moving to LWJGL or waiting for an update from JOGL. I believe @codeanticode is looking into LWJGL.

**Comment by codeanticode** [comment]
@sampottinger indeed, I created a fork from your earlier processing-openjdk repo that uses LWJGL for the core OpenGL renderer:

https://github.com/codeanticode/processing-openjdk/tree/lwjgl-renderer

It still needs some work to encapsulate all AWT dependencies so they are not used when launching a sketch with the P2D or P3D renderers.

**Comment by neilcsmith-net** [comment]
@codeanticode that's not quite correct. There is only a requirement to encapsulate aspects of AWT that can't work headless, is much less.  I have a fork of 3.5.3 with minimal changes to work headless with LWJGL already.  If there's desire to switch upstream I can open an issue and work on a PR for here or your branch.  Mainly a need to push things down into the Surface implementations.  And find a way to execute tasks on the main thread.

**Comment by benfry** [comment]
Should be fixed with the JOGL 2.4 RC that's included with 4.0a1.
